pkg/collector/fake.go

```diff
@@ -0,0 +1,21 @@
+package collector
```

Comment:
fake_test.go ?

Reply:
Agree, but unfortunately, I don't think you can use a  `_test.go` here, as it won't be exported, and therefore not useable from other packages. I have tried it, and a couple of variations to get it working, but if I understand correctly, it's not the way go works. Seems the `*_test.go` files are compiled into the package only when running go test for that package.

Reply:
Can we call it mock collector instead?

Reply:
Kind of followed K8s naming here - all the resources/packages for testing are named "fake", e.g. `k8s.io/client-go/dynamic/fake` or `FakeDiscovery{}` 😉

Reply:
Fine then no worries